scripts/clean_json_file.py

```diff
@@ -13,46 +14,44 @@
 
 
 @click.command()
-@click.argument("filename", type=click.Path(exists=True))
+@click.argument(
+    "filenames", metavar="[FILENAME] ...", type=click.Path(exists=True), nargs=-1
+)
 @click.option(
     "--check",
     is_flag=True,
     default=False,
     help="Don't write the file back, only check whether the file is well formatted.",
 )
-def clean_json_file(filename, check):  # noqa: D301
-    """Clean JSON file.
-
-    Clean JSON file so as to fix indentation, order fields, and remove trailing
-    whitespace.
+def clean_json_file(filenames, check):  # noqa: D301
+    """
+    Clean JSON files.
 
-    \b
-    :param filename: The file name to be reformatted.
-    :param check: Do not modify file, only check whether it is well formatted.
-    :type filename: str
-    :type check: bool
+    Clean one or more JSON files so as to fix indentation, order fields, and
+    remove trailing whitespace.
     """
-    with open(filename, "r") as fdesc:
-        old_content = fdesc.read()
-        records = json.loads(old_content)
-        new_content = (
-            json.dumps(
-                records,
-                indent=2,
-                sort_keys=True,
-                ensure_ascii=False,
-                separators=(",", ": "),
+    for filename in filenames:
+        with open(filename, "r") as fdesc:
+            old_content = fdesc.read()
+            records = json.loads(old_content)
+            new_content = (
+                json.dumps(
+                    records,
+                    indent=2,
+                    sort_keys=True,
+                    ensure_ascii=False,
+                    separators=(",", ": "),
+                )
+                + "\n"
             )
-            + "\n"
-        )
-
-    if old_content != new_content:
-        if check:
-            print(f"[ERROR] File {filename} is badly formatted.")
-            sys.exit(1)
-        else:
-            with open(filename, "w") as fdesc:
-                fdesc.write(new_content)
+
+        if old_content != new_content:
+            if check:
+                print(f"[ERROR] File {filename} is badly formatted.")
+                sys.exit(1)
```

Comment:
Imagine that there are 10 files failing. The first execution of the script will only show the first file, and exit. Then, the user will fix that one, run the script again and it will fail in the next one. 
Wouldn't it be better if it gave the list of all the badly formatted files on the first execution? That would be consistent with the pytest

Reply:
Good point. Fixed.